### DISCUSSION TITLE
Add DiceCELoss as a metric

### DISCUSSION THREAD

**Discussion by matt3o** [Project-MONAI/MONAI#6694] [Ideas]
**Is your feature request related to a problem? Please describe.**
Since DiceCELoss is available as a loss, it would make sense for it to be available as a metric too.

**Describe the solution you'd like**
Either a dedicated DiceCELoss metric like e.g. MeanDice for the DiceLoss OR a pluggable DiceCELossMetric which can be used as input to IgniteMetric.

**Describe alternatives you've considered**
Passing the DiceCELoss into IgniteMetric results in an error since it does not have a reset button. Which kinda makes sense, this is no metric, it does something different.

**Additional context**
I am also curious about why the dice got implemented at two places, namely MeanDice and the DiceLoss. From a quick glance I believe they do the same computation - so is there any reason for the code to duplicated here? The calculation logic for both should be the same.

**Comment by wyli** [comment]
This function actually wraps any loss as part of cumulative metric https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/loss_metric.py perhaps we can focus on this class.

Meandice is not differentiable and (thus) is much faster than dice loss.

**Reply by matt3o** [reply]
@wyli See #6695 about that topic, I tried to implement it with the class you mentioned. Not yet good code, but it's working on my setup at least.

**Reply by matt3o** [reply]
@wyli Just fyi: I did some experiments and actually the loss is 5 times faster for my use case. (label of size num_classes, H > 224, W > 224, D > 224, all calculations on the GPU) 
With `with torch.no_grad(): ` it is  a lot faster than the metric plus it has a is a bit easier to use (Nan workaround with `ignore_empty=False` not needed).

**Comment by wyli** [comment]
addressed in https://github.com/Project-MONAI/MONAI/pull/6695, thank you @matt3o
